src/ansible_navigator/actions/lint.py

```diff
@@ -0,0 +1,372 @@
+                string=obj.path,
+                color=0,
+                decoration=curses.A_BOLD,
+            ),
+        ),
+        (
+            CursesLinePart(
+                column=0,
+                string=check_name,
+                color=severity_to_color(obj.severity),
+                decoration=curses.A_UNDERLINE,
+            ),
+        ),
+    )
+
+
+# TODO: I'd really like this to be a step-level thing. I want to be able to
+# pass a callable to manipulate data (in this case, dig into obj.raw) before
+# it is rendered, without having to do this at an action-global level.
+def filter_content_keys(obj: LintResult) -> Dict[Any, Any]:
+    return obj.raw
+
+
+@actions.register
+class Action(App):
+    """:lint"""
+
+    KEGEX = r"^lint(\s(?P<params>.*))?$"
+
+    def __init__(self, args: ApplicationConfiguration):
+        super().__init__(args=args, logger_name=__name__, name="lint")
+
+    def _fatal(self, msg: str, rc: int = 1) -> Tuple[str, str, int]:
+        self._logger.error(msg)
+
+        if self._args.mode == "interactive":
+            notification = error_notification(messages=[msg])
+            self._interaction.ui.show(notification)
+        else:
+            raise RuntimeError(msg)
+
+        return "", "", rc
+
+    def _run_runner(self) -> Tuple[str, str, int]:
+        """Spin up runner to run ansible-lint, either in an exec env or not."""
+
+        ansible_lint_path = "ansible-lint"
+        if not self._args.execution_environment:
+            ansible_lint_path_maybe = shutil.which("ansible-lint")
+            if ansible_lint_path_maybe is None:
+                err = "'ansible-lint' executable not found, therefore cannot lint."
+                return self._fatal(err, 127)
+            ansible_lint_path = ansible_lint_path_maybe
+
+        kwargs = {
+            "container_engine": self._args.container_engine,
+            "execution_environment_image": self._args.execution_environment_image,
+            "execution_environment": self._args.execution_environment,
+            "navigator_mode": self._args.mode,
+            "pass_environment_variable": self._args.pass_environment_variable,
+            "set_environment_variable": self._args.set_environment_variable,
+            "private_data_dir": self._args.ansible_runner_artifact_dir,
+            "rotate_artifacts": self._args.ansible_runner_rotate_artifacts_count,
+            "timeout": self._args.ansible_runner_timeout,
+            "host_cwd": os.getcwd(),
+            "wrap_sh": True,
+        }
+
+        if isinstance(self._args.execution_environment_volume_mounts, list):
+            kwargs.update(
+                {"container_volume_mounts": self._args.execution_environment_volume_mounts}
+            )
+
+        if isinstance(self._args.container_options, list):
+            kwargs.update({"container_options": self._args.container_options})
+
+        cmd_args = []
+
+        if self._args.mode == "interactive":
```

Comment:
This condition is repeated in a few places. Could you move it into a `@property`?